src/main/java/practice/StreamPractice.java

```diff
@@ -1,80 +1,75 @@
 package practice;
 
-import java.util.Collections;
+import java.util.Arrays;
+import java.util.Collection;
 import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
 import model.Candidate;
+import model.Cat;
 import model.Person;
 
 public class StreamPractice {
-    /**
-     * Given list of strings where each element contains 1+ numbers:
-     * input = {"5,30,100", "0,22,7", ...}
-     * return min integer value. One more thing - we're interested in even numbers.
-     * If there is no needed data throw RuntimeException with message
-     * "Can't get min value from list: < Here is our input 'numbers' >"
-     */
+    private static final CandidateValidator candidateValidator = new CandidateValidator();
+
     public int findMinEvenNumber(List<String> numbers) {
-        return 0;
+        return numbers.stream()
+                .map(el -> el.split(","))
+                .flatMap(Arrays::stream)
+                .map(Integer::parseInt)
+                .filter(el -> el % 2 == 0)
+                .min(Integer::compare)
+                .orElseThrow(() -> new RuntimeException("Can't get min value from list:"
+                        + numbers));
     }
 
-    /**
-     * Given a List of Integer numbers,
-     * return the average of all odd numbers from the list or throw NoSuchElementException.
-     * But before that subtract 1 from each element on an odd position (having the odd index).
-     */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        List<Integer> modifiedList = IntStream.range(0,numbers.size())
+                .mapToObj(i -> i % 2 != 0 ? numbers.get(i) - 1 : numbers.get(i))
+                .toList();
+
+        return modifiedList.stream()
+                .filter(el -> el % 2 != 0)
+                .mapToInt(Integer::intValue)
+                .average()
+                .orElseThrow(NoSuchElementException::new);
```

Comment:
1) Why we collect stream to open it again?
2) Why mapToObj when we may proceed to work with numbers?
```suggestion
        return IntStream.range(0,numbers.size())
                .map(i -> i % 2 != 0 ? numbers.get(i) - 1 : numbers.get(i))
                .filter(el -> el % 2 != 0)
                .average()
                .orElseThrow(NoSuchElementException::new);
```